### PR TITLE
Dialog API buff-up.

### PR DESCRIPTION
**Pull request by DanChadwick**
`parent`, `origin`, `openFrom`, `closeTo` are all optional and all take a jQuery selector or jQuery object.

I found the original API inconsistent and inconvenient in actual use. This PR normalized the API and cleans up some old code remnants. 

Note that `parent` is *not* an id, but a jQuery selector or object. An id is created, if needed, for `{{ember-wormhole}}`.

### CODE REVIEW

**Comment by DanChadwick** [comment]
Test failure seems to be a travis error. It can't install phantomjs for ember-release, but default, beta, and canary all pass fine.

File: addon/components/paper-dialog.js [hunk 1]
```diff
@@ -7,25 +15,40 @@ export default Component.extend({
...
+    }
+    return id;
+  }),
 
   constants: service(),
 
   didInsertElement() {
     if (this.get('escapeToClose')) {
-      let parent = this.get('parentElementSelector');
-      $(parent).on(`keydown.${this.elementId}`, (e) => {
```
Comment (miguelcobain) [inline addon/components/paper-dialog.js]: Why this change?

```diff
+this.get('defaultedParent')
-this.get('parentElementSelector')
```

`defaultedParent` is the new `parentElementSelector`?

File: addon/mixins/translate3d-mixin.js [hunk 2]
```diff
@@ -59,8 +57,10 @@ export default Mixin.create({
       // Wait while CSS takes affect
       // Set the `main` styles and run the transition-in styles
       run.next(() => {
-        this.waitTransitionEnd(this.element).then(() => {
-          this.onTranslateFromEnd();
+        this.waitTransitionEnd($(this.element)).then(() => {
+          if (!this.get('isDestroying') && !this.get('isDestroyed')) {
```
Comment (miguelcobain) [inline addon/mixins/translate3d-mixin.js]: What is the reason for this guard?
For situations where you create and destroy very quickly?
What was the problem you were having?

File: addon/components/paper-dialog.js (hunk 1 above)
Comment (mellatone) [inline addon/components/paper-dialog.js]: +1, `parentElement` || `parentElementSelector` is a very clear descriptor.

Comment (miguelcobain) [inline addon/components/paper-dialog.js]: I'm not complaining about the names. Just trying to understand.
